OWML.ModHelper/ModConfig.cs

```diff
@@ -30,12 +30,20 @@ public T GetSettingsValue<T>(string key)
 
             try
             {
-                var val = value is JObject obj ? obj["value"] : value;
-                return (T)Convert.ChangeType(val, typeof(T));
+                var objectValue = value as JObject;
+                var type = typeof(T);
+                if (type.IsEnum && objectValue != null && objectValue["type"].ToObject<string>() == "selector")
+                {
+                    var selected = (string)objectValue["value"];
+                    var thisEnum = Enum.Parse(type, selected, true);
+                    return Enum.IsDefined(type, thisEnum) ? (T)thisEnum : default;
+                }
+                var val = objectValue != null ? objectValue["value"] : value;
+                return (T)Convert.ChangeType(val, type);
             }
-            catch (InvalidCastException)
+            catch (InvalidCastException ex)
             {
-                ModConsole.Instance.WriteLine($"Error when converting setting {key} of type {value.GetType()} to type {typeof(T)}");
+                ModConsole.Instance.WriteLine($"Error when converting setting {key} of type {value.GetType()} to type {typeof(T)} : {ex.Message};{ex.TargetSite}");
```

Comment:
```suggestion
                ModConsole.Instance.WriteLine($"Error when converting setting {key} of type {value.GetType()} to type {typeof(T)} : {ex}");
```
Converting the exception to string already includes Message and TargetSite (and more stuff).

Reply:
ah, I simply forgot to remove that thing >.<

Reply:
It's much better to include the exception though. There's a few places in OWML that we consume the exception and it makes for hard debugging.

Reply:
in this case it won't tell much AFAIK

Reply:
AFAIK it would simply say "can't convert from source to target" which is even less informative than OWML's error message

Reply:
I see. But it would still have info related to which method threw the error, which is useful.

Reply:
I wasn't able to make it show that :/

Reply:
Oh, I thought Exception.ToString would always do that. Do whatever you please then 🤷‍♂️